app/tasks.py

```diff
@@ -2,66 +2,90 @@
 
 import importlib
 import inspect
+import os
 import sys
+import shutil
 from importlib.metadata import version
 
 from flask import current_app, jsonify, make_response
 
 from app import db
 from app.models import Report, Series
 from hazen import worker
+from celery.utils.log import get_task_logger
 
+logger = get_task_logger(__name__)
 
 @worker.task(bind=True)
-def produce_report(self, user_id, series_id, task_name, image_files, slice_width=None):
+def produce_report(self, user_id, series_id, task_name, image_files,
+                  **kwargs):
+    logger.info("produce report")
     # import Hazen functionality
     task_module = importlib.import_module(f"hazenlib.tasks.{task_name}")
+    logger.info(f"Performing {task_name} task on {series_id}")
 
     # Update Celery task status
     self.update_state(state='PENDING')
 
     # Pass image file path and variables to Hazenlib task
     try:
         task = getattr(task_module, task_name.capitalize())(
-            data_paths=image_files, report=False)  # , report_dir=report_dir
+            input_data=image_files, report=True, **kwargs)
     except:
         class_list = [cls.__name__ for _, cls in inspect.getmembers(
             sys.modules[task_module.__name__],
             lambda x: inspect.isclass(x) and (x.__module__ == task_module.__name__)
             )]
         if len(class_list) == 1:
             task = getattr(task_module, class_list[0])(
-            data_paths=image_files, report=False)  # , report_dir=report_dir
+            input_data=image_files, report=True, **kwargs)
+            logger.info(task)
         else:
             raise Exception(
                 f'Task {task_module} has multiple class definitions: {class_list}')
+
     # Perform task and generate result
-    if task_name == 'snr':
-        result_dict = task.run(slice_width)
-    else:
-        result_dict = task.run()
-    for key, value in result_dict.items():
-        # ignore reports section of the output
-        if key == "reports":
-            pass
-        else:
-            # reconstruct the measurement results
-            result = {key: value}
```

Comment:
no need to handle SNR and other optional args separately, no need to parse the result dict in task-specific way